### PR TITLE
Added read length statistics

### PR DESCRIPTION
**Pull request by rickymagner**
### Description

Added some other statistics relating to the read lengths, with documentation and tests. 

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/analysis/AlignmentSummaryMetrics.java [hunk 1]
```diff
@@ -122,13 +122,31 @@
      */
     public double PF_INDEL_RATE;
 
+    /** The median read length. Computed using all read lengths, not just the aligned ones. */
```
Comment (meganshand) [inline src/main/java/picard/analysis/AlignmentSummaryMetrics.java:125]: Could you rephrase this second sentence to say: `Computed using all read lengths including clipped bases.` and include it in the other metrics you've changed below?

I think that's clearer and I'm basing that off of the difference between the two histograms that are included in AlignmentSummaryMetrics: https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/analysis/AlignmentSummaryMetricsCollector.java#L332

Comment (rickymagner) [reply src/main/java/picard/analysis/AlignmentSummaryMetrics.java]: Made these changes in the following commit!

**Review by gbggrant — APPROVED** [review]

**Review by meganshand — APPROVED** [review]
